storage/imx6ull-flash/flashdev.c

```diff
@@ -135,51 +98,38 @@ static int flashmtd_erase(struct _storage_t *strg, off_t offs, size_t size)
 
 static int flashmtd_read(struct _storage_t *strg, off_t offs, void *data, size_t len, size_t *retlen)
 {
-	int res, err = 0;
+	uint32_t paddr;
+	int err = EOK;
 	size_t tempsz = 0, chunksz;
+	flashdrv_meta_t *meta = strg->dev->ctx->metabuf;
 
-	mutexLock(strg->dev->ctx->lock);
-	if (offs % strg->dev->mtd->writesz) {
-		res = flashdrv_read(strg->dev->ctx->dma, offs / strg->dev->mtd->writesz, strg->dev->ctx->databuf, strg->dev->ctx->metabuf);
-		if (res == flash_uncorrectable) {
-			*retlen = tempsz;
-			mutexUnlock(strg->dev->ctx->lock);
-			return res;
-		}
-
-		err = _flashmtd_errRead(res, strg->dev->ctx->metabuf, err, strg->dev->ctx->databuf);
-		/* TODO: handle error from _flashmtd_errRead */
-
-		chunksz = (strg->dev->mtd->writesz - (offs % strg->dev->mtd->writesz));
-		if (chunksz > len) {
-			chunksz = len;
-		}
-
-		memcpy(data, (unsigned char *)strg->dev->ctx->databuf + (offs % strg->dev->mtd->writesz), chunksz);
-		tempsz += chunksz;
-		offs += chunksz;
-	}
+	paddr = offs / strg->dev->mtd->writesz;
+	offs %= strg->dev->mtd->writesz;
```

Comment:
A small suggestion.
If you use the remainder of the divison as 32 bit number then you reduce the size of the function code over 40 bytes, so for example:
```
uint32_t offs_rem;
offs_rem = offs % strg->dev->mtd->writesz;
```
and futher instead of `offs` use `offs_rem`.
`offs` is 64bit number, `strg->dev->mtd->writesz` is 32bit number, so the remainder from dividing 64bit number by 32bit number is 32 bit number.

Reply:
@Maxez will be unavailable for few days, so something to consider in the future